aerleon/lib/addressbook.py

```diff
@@ -4,12 +4,55 @@
 import itertools
 from typing import List, Union
 
+from aerleon.lib.fqdn import FQDN
 from aerleon.lib.nacaddr import IPv4, IPv6
 
 
 class Addressbook:
     def __init__(self) -> None:
-        self.addressbook = collections.OrderedDict()
+        self._fqdn_addressbook = collections.OrderedDict()
+        self._ip_addressbook = collections.OrderedDict()
+
+    def GetZoneNames(self) -> List:
+        return list(self._ip_addressbook.keys())
+
+    def WalkAddressBook(self, zone_filter=None):
+
+        return self._walk(zone_filter)
+
+    def _walk(self, zone_filter=None):
+        for zone in self._ip_addressbook:
+            if zone_filter and zone != zone_filter:
+                continue
+            groups = set(self._ip_addressbook[zone].keys())
+            groups.update(list(self._fqdn_addressbook[zone].keys()))
+            for group in sorted(groups):
+                ips = []
+                fqdns = []
+                if group in self._ip_addressbook[zone]:
+                    ips = self._ip_addressbook[zone][group]
+                if group in self._fqdn_addressbook[zone]:
+                    fqdns = self._fqdn_addressbook[zone][group]
+                yield zone, group, ips, fqdns
+
+    def GetFQDN(self, zone: str, name: str):
+        return self._fqdn_addressbook[zone][name]
+
+    def GetAddressTokensInZone(self, zone: str):
+        return [i for i in self._ip_addressbook['zone']]
+
+    def GetFQDNTokensInZone(self, zone: str):
+        return [i for i in self._fqdn_addressbook['zone']]
```

Comment:
Looks like a typo. There should probably be a quick test that covers these functions.

```suggestion
    def GetAddressTokensInZone(self, zone: str):
        return [i for i in self._ip_addressbook[zone]]

    def GetFQDNTokensInZone(self, zone: str):
        return [i for i in self._fqdn_addressbook[zone]]
```

Reply:
Also, could this just be `return copy(self._fqdn_addressbook[zone])` ?

Reply:
Done